Mathlib/Data/Matrix/Rank.lean

```diff
@@ -18,17 +18,90 @@ This definition does not depend on the choice of basis, see `Matrix.rank_eq_finr
 ## Main declarations
 
 * `Matrix.rank`: the rank of a matrix
+* `Matrix.cRank`: the rank of a matrix as a cardinal
+* `Matrix.eRank`: the rank of a matrix as a term in `ℕ∞`.
 
 -/
 
-
 open Matrix
 
 namespace Matrix
 
-open Module
+open Module Cardinal Set Submodule
+
+universe um um₀ un un₀ uR uo ul
+variable {m : Type um} {m₀ : Type um₀} {n : Type un} {n₀ : Type un₀} {R : Type uR} {o : Type uo}
+    {l : Type ul}
+
+section Infinite
+
+variable [Semiring R]
+
+/-- The rank of a matrix, defined as the dimension of its column space, as a cardinal. -/
+noncomputable def cRank (A : Matrix m n R) : Cardinal := Module.rank R <| span R <| range Aᵀ
+
+lemma cRank_toNat_eq_finrank (A : Matrix m n R) :
+    A.cRank.toNat = Module.finrank R (span R (range Aᵀ)) := rfl
+
+lemma cRank_mono_col (A : Matrix m n R) (c : n₀ → n) : (A.submatrix id c).cRank ≤ A.cRank := by
+  apply Submodule.rank_mono <| span_mono ?_
+  rintro _ ⟨x, rfl⟩
+  exact ⟨c x, rfl⟩
```

Comment:
Please delete this lemma, it is strictly less general than `cRank_submatrix_le`.

Reply:
Sorry, I thought I had.